src/Raven.Server/Documents/Handlers/Processors/Smuggler/SmugglerHandlerProcessorForExport.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Raven.Client.Documents.Operations;
+using Raven.Server.ServerWide;
+using Raven.Server.ServerWide.Context;
+using Raven.Server.Smuggler.Documents;
+using Raven.Server.Smuggler.Documents.Data;
+using Sparrow.Json;
+
+namespace Raven.Server.Documents.Handlers.Processors.Smuggler
+{
+    internal class SmugglerHandlerProcessorForExport : AbstractSmugglerHandlerProcessorForExport<DatabaseRequestHandler, DocumentsOperationContext>
+    {
+        public SmugglerHandlerProcessorForExport([NotNull] DatabaseRequestHandler requestHandler) : base(requestHandler)
+        {
+        }
+
+        protected override async ValueTask ExportAsync(JsonOperationContext context, long? operationId)
+        {
+            if(operationId.HasValue == false)
+                operationId = RequestHandler.Database.Operations.GetNextOperationId();
+
+            await RequestHandler.Export(context, RequestHandler.DatabaseName, ExportDatabaseInternalAsync, RequestHandler.Database.Operations, operationId.Value);
```

Comment:
Export method should not be in RequestHandler

Reply:
moved to an abstract smuggler processor